docs/source/references/internals/codegen.rst

```diff
@@ -0,0 +1,37 @@
+Pipeline Writing (Creating Code)
+--------------------------------
+
+This section focuses on the internals of generating modular code for TaskDefinitions that can be used by Pipeline Writers.
+For a more high level tutorial on how to write an integration, please check out our integration writing tutorial for contributors.
+
+Definitions
+~~~~~~~~~~~
+
+Here is an illustration of some of the components of a function that lineapy can generate for use in writing Pipelines.
+
+.. image:: ../../_static/images/function_components.png
```

Comment:
The image file is missing (should be committed).

Reply:
woops they were being .gitignored, ty for catching this